lib/analyzer.rb

```diff
@@ -0,0 +1,52 @@
+class Analyzer
+
+    def initialize
+        @vehicle_models = []
+        @most_popular_vehicle = [] 
+    end
+
+    def vehicle_models(vehicles)        
+        vehicles.each do |vehicle|
+            @vehicle_models << vehicle.model
+        end
+        @vehicle_models
+    end
+
+    def model_counter(vehicles)
+        vehicle_models(vehicles)
+        @vehicle_models.tally
+    end
+
+    def most_popular_model(vehicles)
+        most_popular_model = model_counter(vehicles).max_by do |model, count|
```

Comment:
Great use of max_by! No need for the variable though since you're not doing anything with that data other than returning it.